### ISSUE TITLE
Inconsistent role state causing lock-out

### ISSUE THREAD

**Issue by creack** [gravitational/teleport#50296]
Expected behavior:

Valid credentials should let the user login.

If/when deleting a role automatically, teleport should remove the role from the users or allow users to login with a deleted role in their role list.

Current behavior:

When something happens within role validation (like when switching from one version to another), teleport deletes automatically the role but doesn't remove the role from users resulting in the user being locked out.

Also note that the logs are less than helpful. On the backend side it says everything is OK:

```
2024-12-16T13:03:16-06:00 DEBU [AUTH]      User "guillaume.charmes@goteleport.com" authenticated via passwordless auth/methods.go:83
```

On the frontend, it says invalid credentials (while they are valid):

<img width="572" alt="Image" src="https://github.com/user-attachments/assets/10d95a80-9865-4c35-bbed-40d5c4b05f48" />

Bug details:
- Teleport version: 17.0.1
- Recreation steps

Using a different version/branch, create/update a role that would be invalid on master. (Example #50042)
Assign the role to a user, switch back the version/branch and try to login.

**Comment by zmb3** [comment]
This might be intentional behavior (though the UX could use some improvement, and the logs should indicate what the problem was).

If the system says that a user is supposed to have role "foo" and that role cannot be applied (because it is invalid or does not exist), then we have to do the most pessimistic thing and assume that the role had some deny rules. Allowing the user to log in without applying the role could result in elevating their access.

**Comment by creack** [comment]
Agreed, maybe the focus should be on the auto-deletion of the role, either not do that and block/crash, or be explicit that users are in a limbo state.
